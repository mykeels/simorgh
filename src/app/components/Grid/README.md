# Grid

## Usage

When using one of the extended Grid components, e.g. `FrontPageGrid`, if you wish to extend it using the `as` prop, you'll need to use `forwardedAs`. This prop enables you to pass down the element down further than one level.

e.g.
`<Grid forwardedAs="main" role="main ....>...</Grid>`
`<FrontPageGrid forwardedAs="main" role="main ....>...</FrontPageGrid>`
