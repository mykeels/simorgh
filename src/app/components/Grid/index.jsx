import styled from 'styled-components';
import Grid from '@bbc/psammead-grid';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '@bbc/gel-foundations/breakpoints';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_SPACING_QUAD,
} from '@bbc/gel-foundations/spacings';

const group4WrapperMaxWidth = `63rem`; // 1008px
const group5WrapperMaxWidth = `80rem`; // 1280px

export const GridMaxWidthGroup5 = styled(Grid)`
  margin: 0 auto;
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    max-width: ${group4WrapperMaxWidth};
  }
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    max-width: ${group5WrapperMaxWidth};
  }
`;

export const FrontPageGrid = styled(GridMaxWidthGroup5)`
  padding-bottom: ${GEL_SPACING_QUAD};
  margin-top: ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    margin-top: ${GEL_SPACING_DBL};
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    margin-top: 0;
  }
`;

export default Grid;
