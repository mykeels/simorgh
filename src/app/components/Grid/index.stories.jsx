import React from 'react';
import { storiesOf } from '@storybook/react';
import Grid, { GridMaxWidthGroup5 } from '.';

const outerGridProps = {
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 6,
    group5: 6,
  },
};

const gridItemProps = {
  item: true,
  columns: {
    group0: 2,
    group1: 2,
    group2: 2,
    group3: 2,
    group4: 2,
    group5: 2,
  },
};

storiesOf('Components|Grid', module)
  .add('default', () => {
    return (
      <Grid {...outerGridProps}>
        <Grid {...gridItemProps}>
          <p>Item spanning 2 out of 6 columns</p>
        </Grid>
      </Grid>
    );
  })
  .add('Grid as Main', () => {
    return (
      <Grid forwardedAs="main" role="main" {...outerGridProps}>
        <Grid {...gridItemProps}>
          <p>Item with Main element as the parent grid</p>
        </Grid>
      </Grid>
    );
  })
  .add('GridMaxWidthGroup5', () => {
    return (
      <GridMaxWidthGroup5 {...outerGridProps}>
        <Grid {...gridItemProps}>
          <p>Grid item inside parent grid with a defined max-width of group5</p>
        </Grid>
      </GridMaxWidthGroup5>
    );
  });
