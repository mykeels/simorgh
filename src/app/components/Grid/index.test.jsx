import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import Grid, { GridMaxWidthGroup5 } from '.';

const outerGridProps = {
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 6,
    group5: 6,
  },
};

const gridItemProps = {
  item: true,
  columns: {
    group0: 2,
    group1: 2,
    group2: 2,
    group3: 2,
    group4: 2,
    group5: 2,
  },
};

describe('Grid', () => {
  shouldMatchSnapshot(
    'should render Grid with columns',
    <Grid {...outerGridProps}>
      <Grid {...gridItemProps}>
        <p>item spanning 2 out of 6 columns</p>
      </Grid>
    </Grid>,
  );

  shouldMatchSnapshot(
    'should render GridMaxWidthGroup5 with columns',
    <GridMaxWidthGroup5 {...outerGridProps}>
      <Grid {...gridItemProps}>
        <p>Test</p>
      </Grid>
    </GridMaxWidthGroup5>,
  );
});
