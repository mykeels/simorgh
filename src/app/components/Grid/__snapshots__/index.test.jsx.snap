// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render Grid with columns 1`] = `
@supports (display:grid) {
  .c0 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c1 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c1 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c1 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c1 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c1 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c1 {
    display: block;
  }
}

<div
  class="c0"
  dir="ltr"
>
  <div
    class="c1"
    dir="ltr"
  >
    <p>
      item spanning 2 out of 6 columns
    </p>
  </div>
</div>
`;

exports[`Grid should render GridMaxWidthGroup5 with columns 1`] = `
.c1 {
  margin: 0 auto;
}

@supports (display:grid) {
  .c0 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c2 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c2 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c2 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c2 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c2 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c2 {
    width: calc(33.333333333333336%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c2 {
    display: block;
  }
}

@media (min-width:63rem) {
  .c1 {
    max-width: 63rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    max-width: 80rem;
  }
}

<div
  class="c0 c1"
  dir="ltr"
>
  <div
    class="c2"
    dir="ltr"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;
